lib/segment/src/types.rs

```diff
@@ -1699,18 +1701,53 @@ impl PayloadSelector {
         })
     }
 
-    #[allow(clippy::ptr_arg)]
-    pub fn check(&self, key: &PayloadKeyType) -> bool {
-        match self {
-            PayloadSelector::Include(selector) => selector.include.contains(key),
-            PayloadSelector::Exclude(selector) => !selector.exclude.contains(key),
+    pub fn new_exclude(vecs_payload_key_type: Vec<PayloadKeyType>) -> Self {
+        PayloadSelector::Exclude(PayloadSelectorExclude {
+            exclude: vecs_payload_key_type,
+        })
+    }
+
+    fn include_payload_selector(
+        source: &Map<String, Value>,
+        include: &[PayloadKeyType],
+    ) -> Payload {
+        let mut new_json_map = Map::new();
+        for path in include.iter() {
+            let focus = get_value_from_json_map(path, source);
+            match focus {
+                MultiValue::Single(element) => {
+                    if let Some(element) = element {
+                        insert_at_path(path, &mut new_json_map, element.clone());
+                    }
+                }
+                MultiValue::Multiple(elements) => {
+                    let array = elements.iter().cloned().cloned().collect();
```

Comment:
double clone?

Reply:
The iterator is `&&value`, and wants to collect as `value`.

```suggestion
                    let array = elements.into_iter().cloned().collect();
```